### ISSUE TITLE
unexpected value of binary_crossentropy loss function in network with 

### ISSUE THREAD

**Issue by will-cern** [tensorflow/tensorflow#56910]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.8.2

### Custom Code

No

### Current Behaviour?

```shell
Basically I am seeing binary_crossentropy loss evaluate to a completely unexpected number when using it on a network that isn't outputting just a single logit. 

Below is a simple example of a network with two outputs which will correspond to the probability of each class. I wanted to check the calculated loss was what I would expect it to be. All this works fine for the categorical_crossentropy loss function (which is the more natural choice in this case, I accept), but given the code lets you do binary_crossentropy here too I wanted to understand what it's doing. But the meaning of the value seems hard to decipher. Is there a bug or other explanation for the difference in values???
```

### Standalone code to reproduce the issue

```shell
from tensorflow import keras
import numpy as np

loss_func = keras.losses.BinaryCrossentropy()
nn = keras.Sequential([
  keras.layers.Dense(2**8, input_shape=(1,), activation='relu'),
  keras.layers.Dense(2, activation='softmax')
])
nn.compile(loss=loss_func,optimizer='adam')
train_x = np.array([0.4])
train_y = np.array([[0,1]])
print(nn.predict(train_x))
print("Evaluated loss = ",nn.evaluate(train_x,train_y))
print("Function loss = ",loss_func(train_y,nn.predict(train_x)).numpy())
print("Manual loss = ",np.average( -train_y*np.log(nn.predict(train_x)) -(1-train_y)*np.log(1. - nn.predict(train_x)) ))
```
```


### Relevant log output

```shell
[[0.5152152  0.48478484]]
1/1 [==============================] - 0s 92ms/step - loss: 0.7085
Evaluated loss =  0.7084982991218567
Function loss =  0.72405
Manual loss =  0.7240501642227173


The evaluated loss value makes no sense compared to the function value and the manually calculated value (which is just equal to -log(0.48478484) as it should be
```
</details>

**Comment by mohantym** [comment]
Hi @will-cern !
I got same result when the activation function in last layer was changed from 'softmax' to 'sigmoid' and bit different result on the given case. Attached [gist](https://colab.sandbox.google.com/gist/mohantym/8ae63aad2cdf231eb4d2138aa33ac7c2/git_56910.ipynb) for reference.
Thank you!

**Comment by will-cern** [comment]
Hi @mohantym - a sigmoid activation isn't the right thing to be using here though because then the sum of the two network outputs is not equal to 1 i.e. the two outputs are not the respective probabilities of each of the two classes. Furthermore I'm not entirely convinced that they just appear to be agreeing because using the sigmoid activation you get outputs always very close to 0.5 so it looks like the numbers are the same but in fact they aren't.

I think another way to ask my question is this: from the two probabilities of [0.5152152,  0.48478484] (which are the output of the network), what calculation is being performed that results in 0.7084982991218567 ?? 

I can see how to get the number from the "function" call to the loss function, 0.72405, because that's equal to -log(0.48478484). But I cannot figure out what calculation the "evaluate" call is doing to get 0.70849... ???

**Comment by mohantym** [comment]
@will-cern !

Evaluating loss seems close to manual loss and functional loss after reducing the no of neurons in input layers to 64 and adding more hidden layers.
Attached [gist](https://colab.sandbox.google.com/gist/mohantym/06309bf0905cc58c1680d69a75448991/git_56910.ipynb#scrollTo=8MNlfFI3I7-Q) for reference.
Please post in Keras repo for further assistance.

Thank you!

**Comment by jondo** [comment]
(@will-cern in the meantime posted this [in the TF Forum](https://discuss.tensorflow.org/t/how-is-binarycrossentropy-loss-calculated-in-keras-fit-evaluate/11331/) and [in the Keras tracker](https://github.com/keras-team/keras/issues/17125))

**Comment by mohantym** [comment]
Hi @jondo @will-cern !
I suggested to use 64 neurons instead of 2**8 as it will overfit the results.  Was not that a good enough explanation?
Thank you!

**Comment by will-cern** [comment]
That wasn't a satisfactory solution to this issue for me, because the issue is about how the loss function is actually being calculated. Unless I've seriously misunderstood what 'evaluate' does, I believe we should be able to reproduce *exactly* whatever number is calculated by the 'evaluate' method, completely independently of whatever the network architecture is. The loss is simply a function of the network output, so if I take the network output, whatever the output is, I should be able to calculate the loss for myself. 

This does work for other loss functions as far as I can recall, it was just in this particular case with binary cross-entropy on a network with more than 1 output that it is producing something that I cannot verify.
